docs/testspec/src/90_valgrind_sca.rst

```diff
@@ -0,0 +1,107 @@
+.. _valgrind/sca:
+
+Side Channel Analysis Using Valgrind
+=====================================
+
+Overview
+--------
+
+Originally, Valgrind was designed to be a memory error detector. As such it is a
+dynamic analysis tool and analyzes program behaviour during execution. However,
+it can also be used to detect side channel vulnerabilities in cryptographic
+implementations. This approach was first used in `"ctgrind"
+<https://github.com/agl/ctgrind>`_, but the basic idea works with a vanilla
+Valgrind as well.
+
+The implementer adds annotations to mark specific memory regions as explicitly
+secret or explicitly public. Valgrind then tracks runtime data dependencies of
+these regions and warns whenever a control flow decision or indexed memory
+access depend on the annotated secret data or any data that was derived from it.
+
+This is a powerful technique to detect side channel vulnerabilities, but requires
+careful code annotation and a good understanding of the cryptographic algorithm.
+
+Using Valgrind to Detect Side Channels
+--------------------------------------
+
+Secret data regions are marked using ``VALGRIND_MAKE_MEM_UNDEFINED()``, telling
+Valgrind that the data in this region is "undefined". For Valgrind this is
+equivalent to allocated memory that was never explicitly initialized. Note that
+it is legal to read from such regions and even to perform operations on the
+data, as long as no assumption on the result of these operations is made. Hence,
+Valgrind will only emit a warning if such a result is used to create side
+effects that might lead to program bugs. Coincidentally, this is exactly what
+causes certain side channel vulnerabilities based on secret-dependent execution
+patterns.
+
+Explicitly note that the "undefined" state of memory regions is propagated by
+Valgrind as the program performs operations on such values. This means that any
+derived value, such as some operation's result, that is *dependent* on
+"undefined" data will also be *marked* as "undefined".
```

Comment:
Maybe also mention that this detection is not at value-level but bit-level. Not because it's important, but because it's cool 😉

Reply:
Mhm, I agree that it is awesome, but I think the text is dense enough as it stands.